File: app/src/main/java/com/dungnd/android23/recyclerviewadapter.java
package com.dungnd.android23;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class recyclerviewadapter extends RecyclerView.Adapter<recyclerviewadapter.itemholer>{
    ArrayList<FileItem> array;
    public recyclerviewadapter(ArrayList<FileItem> array){
        this.array = array;
    }
    @NonNull
    @Override
    public itemholer onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.itemlayout, parent, false);
        return new itemholer(view);
    }

    @Override
    public void onBindViewHolder(@NonNull itemholer holder, int position) {
        FileItem fileItem = array.get(position);
        holder.textViewfilename.setText(fileItem.getFilename());
    }

    @Override
    public int getItemCount() {
        return array.size();
    }

    public class itemholer extends RecyclerView.ViewHolder {
        TextView textViewfilename;
        public itemholer(@NonNull View itemView) {
            super(itemView);
            textViewfilename = itemView.findViewById(R.id.tv_filename);
        }
    }
}

File: app/src/main/java/com/dungnd/android23/RecyclerViewAdapter.kt
package com.dungnd.android23

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView

class RecyclerViewAdapter(var array: ArrayList<fileitem>) : RecyclerView.Adapter<RecyclerViewAdapter.ItemHolder>() {
    class ItemHolder(itemView: View) : RecyclerView.ViewHolder(itemView){
        var tvtitle : TextView = itemView.findViewById(R.id.tv_title)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemHolder {
        val view : View = LayoutInflater.from(parent.context).inflate(R.layout.itemlayout, parent, false)
        return ItemHolder(view)
    }

    override fun onBindViewHolder(holder: ItemHolder, position: Int) {
        var fileitem = array.get(position)
        holder.tvtitle.text = fileitem.title
    }

    override fun getItemCount(): Int {
        return array.size
    }
}